lib/portlayer/portlayer.go

```diff
@@ -83,5 +84,46 @@ func Init(ctx context.Context, sess *session.Session) error {
 		return err
 	}
 
+	unbindSerialPorts(sess)
+
 	return nil
 }
+
+// unbindSerialPorts disconnects serial ports backed by network on the VCH's old IP
+// for running containers. This is useful when the appliance or the portlayer restarts
+// and the VCH has a new IP. Any errors are logged and portlayer init proceeds as usual.
+func unbindSerialPorts(sess *session.Session) {
+	// Get all running containers from the cache
+	runningState := new(exec.State)
+	*runningState = exec.StateRunning
+	containers := exec.Containers.Containers(runningState)
+
+	for i := range containers {
+		if containers[i].ExecConfig != nil {
+			log.Infof("unbinding serial port for running container: %s", containers[i].ExecConfig.ID)
+		}
+
+		// Obtain a container handle
+		handle := containers[i].NewHandle(context.Background())
+		if handle == nil {
+			log.Error("unable to obtain a handle for container")
```

Comment:
Yes you can as you are checking that in the beginning of the loop and this is init so it's guaranteed to run once before main